instruqt-tracks/terraform-build-aws/setup-our-environment/check-workstation

```diff
@@ -1,10 +1,15 @@
 #!/bin/bash
 set -e
+source /etc/profile.d/instruqt-env.sh
+source /root/.bashrc
 
 # Create /tmp/skip-check to disable this check
 if [ -f /tmp/skip-check ]; then
     rm /tmp/skip-check
     exit 0
 fi
 
-# No check script required here.
\ No newline at end of file
+# Make sure we have valid AWS credentials in our shell environment
+echo $AWS_ACCESS_KEY_ID | grep -P '(?<![A-Z0-9])[A-Z0-9]{20}(?![A-Z0-9])' || fail-message "Something went wrong, we were unable to find a valid AWS_ACCESS_KEY_ID environment variable in your shell. You may need to restart the track."
```

Comment:
Could do something like
```
env | grep AWS_ACCESS_KEY_ID || fail-message "Something went wrong, we were unable to find a valid AWS_ACCESS_KEY_ID environment variable in your shell. You may need to restart the track."
```

...to avoid the complex regex.

Reply:
That wouldn't catch the edge case where `AWS_ACCESS_KEY_ID` is defined, but malformed or empty. I really really don't want students getting past the first challenge without valid, real keys. If anything we should probably even have an API call in here to verify them.